### PR TITLE
TD-631: OpenID Connect Configuration + Default Access Control

### PR DESCRIPTION
**Pull request by weblate-worker**
@vendasta/coding-with-brent 

### Problem

We'd like to use IAM to authenticate users into weblate. IAM supplies configuration for OpenID Connect authorization, and weblate supports OpenID Connect via a base class.

### Solution

Subclass the OpenID Connect base class for Vendasta's configuration. Then add a step to the new user pipeline that assigns users to groups based on their permissions within IAM.

Depends on https://github.com/vendasta/sso/pull/355

### CODE REVIEW

File: weblate/vendasta/access.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    """
+    Add all users to Viewers, add developers to 'Vendasta Internal', and add namespaced users to namespace group.
+    """
+    groups_to_add = [Group.objects.get('Viewers')]
+
+    roles = details.get('roles', [])
+    if 'developer' in roles:
+        groups_to_add.append(Group.objects.get(name='Vendasta Internal'))
+    if 'partner' in roles:
+        groups_to_add.append(Group.objects.get(name='Partner Users'))
```
Comment (rrance-va) [inline weblate/vendasta/access.py:15]: Are developers supposed to be in both the 'Vendasta Internal' and 'Partner Users' roles? (I'm assuming the roles match the personals listed at https://iam-prod.apigateway.co/personas)

File: weblate/vendasta/access.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+def set_permissions(strategy, backend, user, details, **kwargs):
+    """
+    Add all users to Viewers, add developers to 'Vendasta Internal', and add namespaced users to namespace group.
+    """
+    groups_to_add = [Group.objects.get('Viewers')]
+
+    roles = details.get('roles', [])
+    if 'developer' in roles:
+        groups_to_add.append(Group.objects.get(name='Vendasta Internal'))
```
Comment (rrance-va) [inline weblate/vendasta/access.py:13]: Should the group name be 'Vendasta Developers' so it is clear it does not include all employees of Vendasta?

File: weblate/vendasta/access.py (hunk 1 above)
Comment (bperreault-va) [reply weblate/vendasta/access.py]: The keys for personas are listed here https://github.com/vendasta/IAM/blob/master/pkg/subject/persona/persona_types.go

I think it's acceptable for developers to be in both, but it probably doesn't make sense to overlap, given that Vendasta Internal grants access to partner components. I'll add an elif.

File: weblate/vendasta/access.py (hunk 2 above)
Comment (bperreault-va) [reply weblate/vendasta/access.py]: Makes sense! I think I initially meant for this to be any vendastian, but I recognized that didn't necessarily track.

**Pull request merged**